File: Cpp/891_Sum_of_Subsequence_Widths.cpp
class Solution {
public:
    int sumSubseqWidths(vector<int>& A) {
        long result = 0;
        sort(A.begin(), A.end());
        int n = A.size();
        long p = 1;
        for(int i=0; i<n; ++i){
            result = (result + (A[i] - A[n-1-i]) * p) % static_cast<int>(1e9+7);
            p = (p * 2) % static_cast<int>(1e9+7);    
        }
        
        return result;
    }
};

File: Kotlin/891_Sum_of_Subsequence_Widths.kt
class Solution {
    fun sumSubseqWidths(A: IntArray): Int {
        var result = 0L
        A.sort()
        val n = A.size
        var p = 1L
        for(i in 0 until n){
            result = ((result + (A[i] - A[n-1-i]) * p) % (1e9 + 7)).toLong()
            p = ((p * 2) % (1e9 + 7)).toLong()
        }

        return result.toInt()
    }
}